packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -282,7 +289,7 @@ const DataGrid = forwardRef((props, ref) => {
 
     // Set focus to a single header button or hyperlink if they are the only content in cell
     const cellButtonOrHyperlink = focusedCell.querySelector('a, button');
-    if ((isHeaderRow && !focusedCell.hasAttribute('tabindex')) || cellButtonOrHyperlink) {
+    if ((isHeaderRow && !focusedCell.hasAttribute('tabindex')) || (isAutoFocusEnabled && cellButtonOrHyperlink)) {
```

Comment:
will action buttons on header cells still be autofocused, regardless of the prop?

Reply:
The first half of the condition is for the header row and all unit tests pass.  I will manually confirm though.